### DISCUSSION TITLE
list of neighboring ASUs

### DISCUSSION THREAD

**Discussion by DHekstra** [project-gemmi/gemmi#337] [General]
hi, 

we are interested in the following problem:
suppose we have a molecule in an ASU. We'd like to get the list of all molecules that _could_ be in contact with the molecule of interest. it seems that what we want is a list of all ASUs (across symops and translational symmetry of the crystal) that share a face, edge, or vertex with our ASU of interest (the "focal" ASU).

We'd greatly appreciate any insight on how to efficiently get this list of ASUs.

**Comment by wojdyr** [comment]
From what I understand, ASU has different meanings.
I think cctbx has ASU definitions (as shapes – with facets, edges and vertices) in the real space, but that's for density map.
For structures, if ASU is a molecule, ASU shape is defined by the shape of the molecule. The map ASU is not applicable to molecules; a molecule can have a quite different shape and can even span over several unit cells.

You could get a list of ASUs that a molecule is in contact with (i.e. a list of symmetric images of this molecule that are in contact).
Not sure how to define "molecules that *could* be in contact".
